### PR TITLE
Update to webrender 0.10.0

### PR DESCRIPTION
**Pull request by gterzian**
Changes necessary to build servor with webrender 0.10.0

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14286)

### CODE REVIEW

**Comment by gterzian** [comment]
This is a WIP, I didn't see another PR doing the same work, please do let me know if there is one or if anyone is aleady working on this. I am doing this because I wanted to be able to build with `webrender:0.10.0` in order to do some last checks for https://github.com/servo/webrender/pull/533

File: ports/servo/Cargo.toml [hunk 1]
```diff
@@ -53,3 +53,7 @@ sig = "0.1"
 [target.'cfg(target_os = "android")'.dependencies]
 libc = "0.2"
 android_glue = "0.2"
+
+[replace]
```
Comment (emilio) [inline ports/servo/Cargo.toml]: huh, remove this?

**Review by emilio — CHANGES_REQUESTED** [review]

File: ports/servo/Cargo.toml [hunk 2]
```diff
@@ -53,3 +53,7 @@ sig = "0.1"
...
+"webrender:0.10.0" = { path = '/Users/gregory/Projects/webrender/webrender' }
+"webrender_traits:0.10.0" = { path = '/Users/gregory/Projects/webrender/webrender_traits' }
```
Comment (gterzian) [inline ports/servo/Cargo.toml]: sorry this one slipped in, I'll make sure to remove it...

**Comment by emilio** [comment]
I don't know if shaders need to be updated, if not and tests pass, looks good.

**Comment by gterzian** [comment]
@emilio thanks for looking at this, it's still not building locally, I'm in the process of updating `script/dom/webgltexture` to use the latest `webrender_traits::channel::MsgSender`...

File: ports/servo/Cargo.toml (hunk 1 above)
Comment (gterzian) [inline ports/servo/Cargo.toml]: yep :+1:

**Comment by gterzian** [comment]
@emilio in the squash, should I separate the 'update dependencies' and 'update code' stuff, or just put it all in one 'update to webrender 0.10.0' commit?

File: ports/servo/Cargo.toml (hunk 1 above)
Comment (gterzian) [inline ports/servo/Cargo.toml]: this has been done

**Comment by gterzian** [comment]
Any ideas on the windows build failure?

```

error[E0308]: mismatched types
  --> C:\projects\servo\components\gfx\platform\windows\font_template.rs:48:14
   |
48 |         Some(descriptor)
   |              ^^^^^^^^^^ expected struct `dwrote::FontDescriptor`, found a different struct `dwrote::FontDescriptor`
   |
   = note: expected type `dwrote::FontDescriptor` (struct `dwrote::FontDescriptor`)
   = note:    found type `dwrote::FontDescriptor` (struct `dwrote::FontDescriptor`)
note: Perhaps two different versions of crate `dwrote` are being used?
  --> C:\projects\servo\components\gfx\platform\windows\font_template.rs:48:14
   |

```

**Comment by emilio** [comment]
Yep, that's a duplicated dependency version, you need to update both of 
them.

**Comment by gterzian** [comment]
@emilio thanks! Could you please elaborate on how to update them? FYI thy are both at the up to date version, but one is pointing to Github, while the other points to cargo...

**Comment by jdm** [comment]
Sounds like we should change the one that's pointing at git to use cargo instead.

**Comment by gterzian** [comment]
Also, from manual testing I was getting just a flickering black screen previously, will test again tomorrow, wondering if perhaps more changes are required then the minimum needed to successfully compile...

**Comment by glennw** [comment]
@gterzian Thanks for working on this! I've pushed 3 new commits to your PR branch. The first one updates the display list generation for scroll API changes. The 2nd one enables the IPC feature of WR, and switches dwrote to the crates.io version. The 3rd commit removes all the old shaders that are now embedded by default.

With these changes, it's running locally on my machine fine (in both the WPT tests and some manual website tests).

**Comment by glennw** [comment]
@bors-servo retry
 * Seeing which are repeatable, as none of these timeouts are reproducible locally.

**Comment by glennw** [comment]
These new timeouts seem permanent, but I can't reproduce them locally (I'm fairly confident they are unrelated to the WR update, but I'll see if I can repro on the build machines).

**Comment by glennw** [comment]
@bors-servo retry
 * #14215
 * #14217

**Comment by gterzian** [comment]
@glennw thanks for your help!

**Comment by glennw** [comment]
@bors-servo retry
* Infra?

**Comment by glennw** [comment]
The commits by @gterzian look fine to me. r? anyone else on my commits added to this PR?

I expect this will still have some new intermittent failures (which are not locally reproducible for me). I'm fairly confident they aren't related to this PR at all, but I'll do some investigating today on the build machines. It should still be fine to review now though.

**Comment by glennw** [comment]
@bors-servo retry
 * https://github.com/servo/servo/issues/12797

**Comment by glennw** [comment]
@bors-servo retry
 * #12797
 * #14267

**Comment by glennw** [comment]
@bors-servo retry
 * #12797 
 * #13555

**Comment by jdm** [comment]
@bors-servo: retry
* #12797 #14319

**Comment by jdm** [comment]
I'm really not sure what to do here. This WR update is clearly stressing the build machines in ways that other PRs do not currently, which is causing havoc in a bunch of tests that appear to be timing-dependent.

**Comment by glennw** [comment]
@jdm Yea, I'm similarly unsure what to do here. This PR will definitely change timing dependent things (e.g. shaders are now embedded in binary and not loaded from disk), but I'm fairly confident all the intermittents that it triggers are not an issue with this PR itself, but exposing other bugs :( I'm going to just spend all of today trying to track down the root cause of some of these intermittents. Perhaps there's only 1-2 things causing most of these to trigger.

**Comment by glennw** [comment]
@bors-servo retry
 * #13776

**Comment by kvark** [comment]
@gterzian heads up! https://github.com/servo/webrender/pull/591 has supposedly fixed the intermittent issues with the bots. So if you update the PR to the latest WR, it has a chance of passing through.

**Pull request merged**